torch/distributed/numa_binding.py

```diff
@@ -0,0 +1,528 @@
+"""
+This script is used for binding the rank-process to cpu-cores.
+The current binding options are as follows:
+1. Node (node)
+2. Socket (called socket)
+3. Exclusive (called exclusive)
+4. Core-Complex (Called core-complex)
+"""
+import subprocess
+import os
+import socket
+from argparse import ArgumentParser, REMAINDER
+from collections import defaultdict
+import re
+import shutil
+import logging
+import pynvml
+
+# Commands/Paths to get system-level information
+NUMA_CMD = "/sys/bus/pci/devices/{value}/numa_node"
+NUMA_CPU_MAP_CMD = "/sys/devices/system/node/node{value}/cpumap"
+CPU_MAP_CMD = "/proc/self/status | grep \"Cpus_allowed:\""
+THREAD_SIBLINGS_CMD = "/sys/devices/system/cpu/cpu{value}/topology/thread_siblings"
+SHARED_CACHE_CMD = "/sys/devices/system/node/node{value_1}/cpu{value_2}/cache/{value_3}/shared_cpu_map"
+CACHE_CMD = "/sys/devices/system/node/node{value_1}/cpu{value_2}/cache"
+SOCKET_CMD = "/sys/devices/system/node/node{value}/cpulist"
+PHYSICAL_PACKAGE_ID_CMD = "/sys/devices/system/cpu/cpu{value}/topology/physical_package_id"
+POSSIBLE_NODES_CMD = "/sys/devices/system/node/possible"
+
+logging.basicConfig(level=logging.INFO)
+
+class System:
+    """
+    Abstracts system specific methods, so it could be mocked for unit tests, across various different types
+    of systems.
+    """
+
+    def __init__(self):
+        pass
+
+    def execute_command(self, cmd):
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True)
+        outs, errs = process.communicate()
+        if process.returncode != 0:
+            raise Exception(f"command:{cmd}, return code:{process.returncode},stderr:{errs}")
+        return outs.decode('UTF-8')
+
+    # returns number of gpus
+    def get_gpu_count(self):
+        # Initialize NVML
+        pynvml.nvmlInit()
+        # Get the number of GPU devices
+        device_count = pynvml.nvmlDeviceGetCount()
+        # Shutdown NVML
+        pynvml.nvmlShutdown()
```

Comment:
Is there a device-generic way? 
There should be some methods in `torch.accelerator` package now.